lib/std/decode_helpers.nim

```diff
@@ -1,13 +1,11 @@
-# Include file that implements 'decodePercent' and friends. Do not import it!
-
-proc handleHexChar(c: char, x: var int, f: var bool) {.inline.} =
+proc handleHexChar*(c: char, x: var int, f: var bool) {.inline.} =
   case c
   of '0'..'9': x = (x shl 4) or (ord(c) - ord('0'))
   of 'a'..'f': x = (x shl 4) or (ord(c) - ord('a') + 10)
   of 'A'..'F': x = (x shl 4) or (ord(c) - ord('A') + 10)
```

Comment:
lexer.handleHexChar diverges here:
```nim
  of '"', '\'':
    if position <= 1: invalid()
    # do not progress the bufpos here.
    if position == 0: inc(L.bufpos)
```
how will this be reconciled?

Reply:
I am not sure.
maybe
```nim
case c:
of '0' .. '9', 'a' .. 'f': handleHexChar(c)
else: ...
```
Or don't change that